src/Recurr/Rule.php

```diff
@@ -253,6 +253,39 @@ public static function createFromArray(
     }
 
     /**
+     * Create a Rule object based on natural language text.
+     *
+     * @param string $text Natural language description like "every day for 3 times"
+     * @param \DateTime|\DateTimeImmutable|string|null $startDate Start date for the recurrence
+     * @param \DateTime|\DateTimeImmutable|string|null $endDate End date for the recurrence
+     * @param string|null $timezone Timezone identifier
+     *
+     * @return self
+     *
+     * @throws InvalidRRule If the text cannot be parsed
+     */
+    public static function createFromText(
+        string $text,
+        \DateTime|\DateTimeImmutable|string|null $startDate = null,
+        \DateTime|\DateTimeImmutable|string|null $endDate = null,
+        ?string $timezone = null,
+    ): self {
+        require_once __DIR__ . '/TextParser.php';
```

Comment:
Shouldn't be necessary with Composer's autoloader

Reply:
Removed